File: horizons_ts/src/evaluation.ts
import { HorizonsClient } from "./client";
import { EvalReportRow, UUID } from "./types";

export class EvaluationAPI {
  constructor(private client: HorizonsClient) {}

  async run(casePayload: Record<string, unknown>, project_id?: UUID): Promise<EvalReportRow> {
    const body: Record<string, unknown> = { case: casePayload };
    if (project_id) body.project_id = project_id;
    return await this.client.request<EvalReportRow>("/api/v1/eval/run", {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(body),
    });
  }

  async reports(params: { project_id?: UUID; report_id?: UUID; limit?: number; offset?: number } = {}): Promise<Record<string, unknown>> {
    const search = new URLSearchParams();
    Object.entries(params).forEach(([k, v]) => {
      if (v !== undefined && v !== null) search.append(k, String(v));
    });
    const path = `/api/v1/eval/reports${search.toString() ? `?${search.toString()}` : ""}`;
    return await this.client.request<Record<string, unknown>>(path, { method: "GET" });
  }
}

File: horizons_py/horizons/evaluation.py
from __future__ import annotations

from typing import Any, Dict, Optional
from uuid import UUID

from .client import HorizonsClient
from . import models


class EvaluationAPI:
    def __init__(self, client: HorizonsClient) -> None:
        self._client = client

    async def run(self, *, case: Dict[str, Any], project_id: Optional[UUID] = None) -> models.EvalReportRow:
        body: Dict[str, Any] = {"case": case}
        if project_id:
            body["project_id"] = str(project_id)
        resp = await self._client._request("POST", "/api/v1/eval/run", json=body)
        data = await self._client.json(resp)
        return models.EvalReportRow.model_validate(data)

    async def reports(
        self,
        *,
        project_id: Optional[UUID] = None,
        report_id: Optional[UUID] = None,
        limit: int = 50,
        offset: int = 0,
    ) -> Dict[str, Any]:
        params: Dict[str, Any] = {"limit": limit, "offset": offset}
        if project_id:
            params["project_id"] = str(project_id)
        if report_id:
            params["report_id"] = str(report_id)
        resp = await self._client._request("GET", "/api/v1/eval/reports", params=params)
        return await self._client.json(resp)
